File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/annotation/ValidateUsing.java
package com.wesleyelliott.kubwa.annotation;


import com.wesleyelliott.kubwa.rule.Rule;

import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

/**
 * Created by wesley on 2016/07/28.
 */

@Retention(RUNTIME)
@Target(ANNOTATION_TYPE)
public @interface ValidateUsing {
    Class<? extends Rule> value();
}

File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/annotation/ValidateUsing.kt
package com.wesleyelliott.kubwa.annotation


import com.wesleyelliott.kubwa.rule.Rule

import kotlin.reflect.KClass

/**
 * Created by wesley on 2016/07/28.
 */

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.ANNOTATION_CLASS)
annotation class ValidateUsing(val value: KClass<out Rule<*>>)